misoc/software/common.mak

```diff
@@ -59,6 +63,9 @@ export CFLAGS_$(subst -,_,$(CARGO_TRIPLE)) = $(CFLAGS)
 
 # Linker options
 LDFLAGS = --gc-sections -nostdlib -nodefaultlibs -L$(BUILDINC_DIRECTORY)
+ifeq ($(CPU),vexriscv)
+	LDFLAGS += -nostartfiles
```

Comment:
Why is this needed for risc-v but not the other CPUs?

Reply:
From what i have seen for RISC-V, when you want do to baremetal, you add -nostartfiles  -nostdlib  as compiler flags. 
After some try, it look like that if you are not using the std libs, it will not try to add a startup file. But it's better to not play with the fire. 

I have no idea for other CPU. It probably depend how far they implemented the clib and stuff like this. But i'm not a c guru. Maybe i'm wrong.

Reply:
I don't think this can really go wrong in obscure ways... if it still works without ``-nostartfiles`` I think those three lines should be removed.

Reply:
`-nodefaultlibs` implies `-nostartfiles`.

Reply:
From https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html

-nodefaultlibs
 ... The standard startup files are used normally, unless -nostartfiles is used.

But -nostdlib implies -nostartfiles (from the same link)

So great, i removed it :D